### PR TITLE
feat(system-e2e): Move system-e2e to NX

### PR DESCRIPTION
**Pull request by AndesKrrrrrrrrrrr**
Same as #13968, but don't touch CI. Added #14245.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Added support for Playwright in VS Code and system E2E tests.
	- Introduced new Playwright test configurations and environment setups.
	- Updated end-to-end testing commands and instructions in documentation.
	- Enhanced debugging capabilities and URL management in tests.

- **Bug Fixes**
	- Improved element visibility checks and text content validation in tests.
	- Adjusted page load state checks for better synchronization during tests.

- **Documentation**
	- Updated `README.md` with new test execution commands and examples.

- **Chores**
	- Updated CI scripts for better formatting and flexibility.
	- Enhanced cleaning scripts with additional functionality.

- **Refactor**
	- Streamlined TypeScript configurations and simplified script commands.
	- Refactored test files for clearer interaction and validation logic.

- **Tests**
	- Added and updated various test cases across the system, focusing on clarity and localization improvements.

### CODE REVIEW

File: scripts/clean.sh [hunk 1]
```diff
@@ -3,6 +3,7 @@
 set -euo pipefail
 
 CLEAN_DRY=false
+CLEAN_DIST=false
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply scripts/clean.sh]: Changes here have been moved to another PR.

File: apps/system-e2e/README.md [hunk 2]
```diff
@@ -6,8 +6,8 @@
 
 System smoke tests are run against our live [dev web](https://beta.dev01.devland.is/), on every commit to `main`.
 This means that changes done on the dev web will be reflected in the system tests.
-E.g. if you change a users preferred locale, your test will break if it's not language-agnostic.
-To combat this we have implemented `disablers.disable*` and `urls.icelandicAndNoPopup`.
+E.g. if you change a user's preferred locale, your test will break if it's not language-agnostic.
+To combat this, we have implemented `disablers.disable*` and `urls.icelandicAndNoPopup`.
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:10]: Already present.

File: apps/system-e2e/README.md [hunk 3]
```diff
@@ -6,8 +6,8 @@ Welcome to testing. For your convenience each section has a TL;DR, so you can qu
 
 System smoke tests are run against our live [dev web](https://beta.dev01.devland.is/), on every commit to `main`.
 This means that changes done on the dev web will be reflected in the system tests.
-E.g. if you change a users preferred locale, your test will break if it's not language-agnostic.
-To combat this we have implemented `disablers.disable*` and `urls.icelandicAndNoPopup`.
+E.g. if you change a user's preferred locale, your test will break if it's not language-agnostic.
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:9]: Already fixed

[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:9]: Indeed, but has been fixed now

**Comment by AndesKrrrrrrrrrrr** [comment]
@coderabbitai review

File: apps/system-e2e/README.md [hunk 4]
```diff
@@ -6,8 +6,8 @@ Welcome to testing. For your convenience each section has a TL;DR, so you can qu
...
+To combat this, we have implemented `disablers.disable*` and `urls.icelandicAndNoPopup`.
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:10]: There is nothing to correct

File: apps/system-e2e/README.md [hunk 5]
```diff
@@ -125,7 +125,7 @@
 - Hotel permit with food, but no alcohol
 - Hotel permit with food and alcohol
 - Bar with only alcohol
-- Home accommodation (AirBnB style), no food, no alcohol
+- Home accommodation (Airbnb style), no food, no alcohol
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:128]: We want British English convention.

File: apps/system-e2e/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:10]: Never mind, the word "out" was missing.

File: apps/system-e2e/README.md [hunk 6]
```diff
@@ -177,38 +220,54 @@
...
+This is an infrastructure issue which should have been resolved. If you see this
+in your tests on dev, contact DevOps 👩‍💻
 
-## ⌛ Tests are timing out
+### ⌛ Tests are timing out
 
-This indicates either a local network issue like bad internet (your problem), or a performance problem in the application code. In some cases this can be remedied by increasing the timeout if a certain site/service is known to be slow with `page.goto('/my-url', { timeout: Timeout.medium })`.
+This indicates either a local network issue like bad internet (your problem), or
+a performance problem in the application code. In some cases, this can be remedied
+by increasing the timeout if a certain site/service is known to be slow with
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:272]: This does not adhere to our code standards

**Review by robertaandersen — APPROVED** [review]
LGTM ( a lot in here :) - Maybe this could have been broken down to smaller requests for clarity)

File: apps/system-e2e/README.md [hunk 7]
```diff
@@ -177,38 +220,54 @@
...
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:272]: This would fail our formatting check. Ignoring.

**Review by robertaandersen — APPROVED** [review]
LGTM

File: package.json [hunk 8]
```diff
@@ -442,8 +446,11 @@
     "mailparser": "3.5.0",
     "next-secure-headers": "2.1.0",
     "nx": "16.9.0",
+    "playwright": "1.42.0",
+    "playwright-tesults-reporter": "^1.0.1",
     "prettier": "2.6.2",
     "sequelize-cli": "6.4.1",
+    "slack-notify": "^2.0.3",
```
Comment (saevarma) [inline package.json:453]: Pin dependency version

File: package.json [hunk 9]
```diff
@@ -341,13 +343,14 @@
     "@nx/nest": "16.9.0",
     "@nx/next": "16.9.0",
     "@nx/node": "16.9.0",
+    "@nx/playwright": "16.9",
```
Comment (saevarma) [inline package.json:346]: We should set the version to `16.9.0` to match the other nx package version.

File: package.json (hunk 8 above)
Comment (AndesKrrrrrrrrrrr) [reply package.json:453]:
```suggestion
    "playwright-tesults-reporter": "1.0.1",
    "prettier": "2.6.2",
    "sequelize-cli": "6.4.1",
    "slack-notify": "2.0.3",
```

File: package.json (hunk 9 above)
Comment (AndesKrrrrrrrrrrr) [reply package.json:346]:
```suggestion
    "@nx/playwright": "16.9.0",
```

File: apps/system-e2e/README.md [hunk 10]
```diff
@@ -39,49 +43,66 @@ For local development and testing start your app. Generally, first-time setup an
...
+- Specific test file: `yarn system-e2e -g '<path/to/your/test/file>'`
+- Using a path pattern (regex): `yarn system-e2e -g '<pattern>'`
 
 {% hint style="example" %}
 
-- smoke: `yarn system-e2e 'application-system-form/smoke'`
-- acceptance: `yarn system-e2e 'service-portal/acceptance'`
-- both: `yarn system-e2e 'system-e2e/.*/web'`
-- pattern `yarn system-e2e 'system-e2e/.*/s?port?'`
+- smoke: `yarn system-e2e -g 'application-system-form/smoke'`
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:63]: This is a playwright flag passed through. Please suggest some documentation.

[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:63]: This is a standard playwright nx executor flag, meaning `grep`, which searches for files to test on. Do we not mention that the `yarn system-e2e` is simply a light wrapper`? If so, dismiss your concerns.

File: apps/system-e2e/README.md [hunk 11]
```diff
@@ -39,49 +43,69 @@ For local development and testing start your app. Generally, first-time setup an
...
```
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:66]: This is correct as is, and is documented in the file.

[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:66]: Please reconsider your requested changes

**Review by saevarma — APPROVED** [review]
🚀

File: apps/system-e2e/README.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (AndesKrrrrrrrrrrr) [reply apps/system-e2e/README.md:66]: I pushed a fix in 95c34f8

**Comment by AndesKrrrrrrrrrrr** [comment]
@coderabbitai review

**Pull request merged**